### DISCUSSION TITLE
How to calculate the number of unbranched non-ring chains with RDKIT?

### DISCUSSION THREAD

**Discussion by chengthefang** [rdkit/rdkit#5012] [Q&A]
Hi All,

Recently, I found an interesting topological descriptor, "Number of unbranched non-ring chains" that can be calculated by Accelrys Pipeline Pilot or OpenEye. I wonder if there is a way to calculate it with RDKIT.

Some reference compounds would be
Smiles                                                                                                                Number of unbranched non-ring chains
CC(N)Cc1ccccc1                                                                                                                               2
CC1CC2C3CCC4=CC(=O)C=CC4(C3(C(CC2(C1(C(=O)CO)O)C)O)F)C                                     9
NC(=O)CS(=O)C(c1ccccc1)c2ccccc2                                                                                             4

Any thoughts?

Thanks,
Cheng

**Comment by greglandrum** [comment]
@chf42 Do you have an pointer to how this is defined? Based on the first three molecules, I thought I might understand what the definition of this descriptor is, but the last one confuses me.

I'm also curious what the value is for the molecule CCC(C)CCC

**Reply by chengthefang** [reply]
> @chf42 Do you have an pointer to how this is defined? Based on the first three molecules, I thought I might understand what the definition of this descriptor is, but the last one confuses me.
> 
> I'm also curious what the value is for the molecule CCC(C)CCC

Hi Greg,

Unfortunately, I didn't know how the unbranched chain was defined although I knew it was calculated with Pipeline Pilot. I am referring to this paper "Technically Extended MultiParameter Optimization (TEMPO): An Advanced Robust Scoring Scheme To Calculate Central Nervous System Druggability and Monitor Lead Optimization". In Table2, they calculated different properties for 5 molecules using PP or Schrodinger Qikprop.  Due to no access to PP, I couldn't get more detailed description of NChains. 

I guess the value for CCC(C)CCC should be 2.

BTW, I also found the inconsistency in calculated values of HBA and NRotatableBonds when using RDKIT and PP. I wonder if you are familiar with the difference in the definition of HBA and NRotatableBonds in RDKIT compared to PP. Is it possible to reproduce PP-calculated values with RDKIT?

Thanks,
Cheng

**Reply by greglandrum** [reply]
> BTW, I also found the inconsistency in calculated values of HBA and NRotatableBonds when using RDKIT and PP. I wonder if you are familiar with the difference in the definition of HBA and NRotatableBonds in RDKIT compared to PP. Is it possible to reproduce PP-calculated values with RDKIT?

For descriptors where there's no a priori correct answer like HBA, HBD, NumRotatableBonds, etc, we try to provide reasonable and consistent values; not to reproduce the values provided by some other piece of software.

If you can figure out what the PP definitions are, you can definitely write your own calculator to try and reproduce those values, but that's not something which we would make part of the core RDKit.

**Reply by greglandrum** [reply]
> Unfortunately, I didn't know how the unbranched chain was defined although I knew it was calculated with Pipeline Pilot.

Without a clear definition of a descriptor, there's no way to implement it. "Number of chains" (which is what is used in the TEMPO paper) or "number of unbranched non-ring chains" don't seem to have unique and unambiguous definitions, so even if we were to invent something there, it's very unlikely that the results would be the same as what PP generates.

**Reply by chengthefang** [reply]
> > Unfortunately, I didn't know how the unbranched chain was defined although I knew it was calculated with Pipeline Pilot.
> 
> Without a clear definition of a descriptor, there's no way to implement it. "Number of chains" (which is what is used in the TEMPO paper) or "number of unbranched non-ring chains" don't seem to have unique and unambiguous definitions, so even if we were to invent something there, it's very unlikely that the results would be the same as what PP generates.

Yes, Greg. I totally agree with you. The Number of chains doesn't has a unique definition. Thank you for your comments!

**Reply by chengthefang** [reply]
> > BTW, I also found the inconsistency in calculated values of HBA and NRotatableBonds when using RDKIT and PP. I wonder if you are familiar with the difference in the definition of HBA and NRotatableBonds in RDKIT compared to PP. Is it possible to reproduce PP-calculated values with RDKIT?
> 
> For descriptors where there's no a priori correct answer like HBA, HBD, NumRotatableBonds, etc, we try to provide reasonable and consistent values; not to reproduce the values provided by some other piece of software.
> 
> If you can figure out what the PP definitions are, you can definitely write your own calculator to try and reproduce those values, but that's not something which we would make part of the core RDKit.

Agree here as well. Since different piece of software might generate different values,  the generation of reasonable and consistent values across molecules will make more sense.
